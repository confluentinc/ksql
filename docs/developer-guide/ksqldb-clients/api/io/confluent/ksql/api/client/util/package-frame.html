<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_265) on Fri Oct 01 11:59:32 CDT 2021 -->
<title>io.confluent.ksql.api.client.util</title>
<meta name="date" content="2021-10-01">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../../../../../script.js"></script>
</head>
<body>
<h1 class="bar"><a href="../../../../../../io/confluent/ksql/api/client/util/package-summary.html" target="classFrame">io.confluent.ksql.api.client.util</a></h1>
<div class="indexContainer">
<h2 title="Classes">Classes</h2>
<ul title="Classes">
<li><a href="ClientTestUtil.html" title="class in io.confluent.ksql.api.client.util" target="classFrame">ClientTestUtil</a></li>
<li><a href="ClientTestUtil.TestSubscriber.html" title="class in io.confluent.ksql.api.client.util" target="classFrame">ClientTestUtil.TestSubscriber</a></li>
<li><a href="JsonMapper.html" title="class in io.confluent.ksql.api.client.util" target="classFrame">JsonMapper</a></li>
<li><a href="RowUtil.html" title="class in io.confluent.ksql.api.client.util" target="classFrame">RowUtil</a></li>
<li><a href="RowUtilTest.html" title="class in io.confluent.ksql.api.client.util" target="classFrame">RowUtilTest</a></li>
</ul>
</div>
</body>
</html>
