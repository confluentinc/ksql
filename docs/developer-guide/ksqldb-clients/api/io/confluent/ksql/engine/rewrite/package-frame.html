<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (1.8.0_265) on Fri Oct 01 11:59:33 CDT 2021 -->
<title>io.confluent.ksql.engine.rewrite</title>
<meta name="date" content="2021-10-01">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
<script type="text/javascript" src="../../../../../script.js"></script>
</head>
<body>
<h1 class="bar"><a href="../../../../../io/confluent/ksql/engine/rewrite/package-summary.html" target="classFrame">io.confluent.ksql.engine.rewrite</a></h1>
<div class="indexContainer">
<h2 title="Classes">Classes</h2>
<ul title="Classes">
<li><a href="AstSanitizer.html" title="class in io.confluent.ksql.engine.rewrite" target="classFrame">AstSanitizer</a></li>
<li><a href="AstSanitizerTest.html" title="class in io.confluent.ksql.engine.rewrite" target="classFrame">AstSanitizerTest</a></li>
<li><a href="DataSourceExtractorTest.html" title="class in io.confluent.ksql.engine.rewrite" target="classFrame">DataSourceExtractorTest</a></li>
<li><a href="ExpressionTreeRewriter.html" title="class in io.confluent.ksql.engine.rewrite" target="classFrame">ExpressionTreeRewriter</a></li>
<li><a href="ExpressionTreeRewriter.Context.html" title="class in io.confluent.ksql.engine.rewrite" target="classFrame">ExpressionTreeRewriter.Context</a></li>
<li><a href="ExpressionTreeRewriterTest.html" title="class in io.confluent.ksql.engine.rewrite" target="classFrame">ExpressionTreeRewriterTest</a></li>
<li><a href="QueryAnonymizer.html" title="class in io.confluent.ksql.engine.rewrite" target="classFrame">QueryAnonymizer</a></li>
<li><a href="QueryAnonymizerTest.html" title="class in io.confluent.ksql.engine.rewrite" target="classFrame">QueryAnonymizerTest</a></li>
<li><a href="StatementRewriteForMagicPseudoTimestamp.html" title="class in io.confluent.ksql.engine.rewrite" target="classFrame">StatementRewriteForMagicPseudoTimestamp</a></li>
<li><a href="StatementRewriteForMagicPseudoTimestampTest.html" title="class in io.confluent.ksql.engine.rewrite" target="classFrame">StatementRewriteForMagicPseudoTimestampTest</a></li>
<li><a href="StatementRewriter.html" title="class in io.confluent.ksql.engine.rewrite" target="classFrame">StatementRewriter</a></li>
<li><a href="StatementRewriter.Context.html" title="class in io.confluent.ksql.engine.rewrite" target="classFrame">StatementRewriter.Context</a></li>
<li><a href="StatementRewriterTest.html" title="class in io.confluent.ksql.engine.rewrite" target="classFrame">StatementRewriterTest</a></li>
</ul>
</div>
</body>
</html>
