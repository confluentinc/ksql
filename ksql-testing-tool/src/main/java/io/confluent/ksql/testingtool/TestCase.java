/*
 * Copyright 2019 Confluent Inc.
 *
 * Licensed under the Confluent Community License (the "License"); you may not use
 * this file except in compliance with the License.  You may obtain a copy of the
 * License at
 *
 * http://www.confluent.io/confluent-community-license
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations under the License.
 */

package io.confluent.ksql.testingtool;

import static org.hamcrest.Matchers.is;
import static org.junit.Assert.assertThat;
import static org.junit.Assert.fail;
import static org.junit.matchers.JUnitMatchers.isThrowable;

import com.google.common.collect.ImmutableMap;
import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
import io.confluent.ksql.metastore.MetaStore;
import io.confluent.ksql.services.ServiceContext;
import io.confluent.ksql.util.KsqlConstants;
import java.nio.file.Path;
import java.util.Collection;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Objects;
import java.util.Optional;
import org.apache.kafka.clients.producer.ProducerRecord;
import org.apache.kafka.common.serialization.StringDeserializer;
import org.apache.kafka.streams.test.ConsumerRecordFactory;
import org.apache.kafka.streams.test.OutputVerifier;
import org.hamcrest.StringDescription;

public class TestCase implements Test {

  private final Path testPath;
  private final String name;
  private final Map<String, Object> properties;
  private final Collection<Topic> topics;
  private final List<Record> inputRecords;
  private final List<Record> outputRecords;
  private final List<String> statements;
  private final ExpectedException expectedException;
  private List<String> generatedTopology;
  private List<String> generatedSchemas;
  private Optional<TopologyAndConfigs> expectedTopology = Optional.empty();
  private final PostConditions postConditions;

  @Override
  public String getName() {
    return name;
  }

  @Override
  public String getTestFile() {
    return testPath.toString();
  }

  TestCase(
      final Path testPath,
      final String name,
      final Map<String, Object> properties,
      final Collection<Topic> topics,
      final List<Record> inputRecords,
      final List<Record> outputRecords,
      final List<String> statements,
      final ExpectedException expectedException,
      final PostConditions postConditions
  ) {
    this.topics = topics;
    this.inputRecords = inputRecords;
    this.outputRecords = outputRecords;
    this.testPath = testPath;
    this.name = name;
    this.properties = ImmutableMap.copyOf(properties);
    this.statements = statements;
    this.expectedException = expectedException;
    this.postConditions = Objects.requireNonNull(postConditions, "postConditions");
  }

  TestCase copyWithName(final String newName) {
    final TestCase copy = new TestCase(
        testPath,
        newName,
        properties,
        topics,
        inputRecords,
        outputRecords,
        statements,
        expectedException,
        postConditions);
    copy.generatedTopology = generatedTopology;
    copy.expectedTopology = expectedTopology;
    copy.generatedSchemas = generatedSchemas;
    return copy;
  }

  void setGeneratedTopology(final List<String> generatedTopology) {
    this.generatedTopology = Objects.requireNonNull(generatedTopology, "generatedTopology");
  }

  void setExpectedTopology(final TopologyAndConfigs expectedTopology) {
    this.expectedTopology = Optional.of(expectedTopology);
  }

  void setGeneratedSchemas(final List<String> generatedSchemas) {
    this.generatedSchemas = Objects.requireNonNull(generatedSchemas, "generatedSchemas");
  }

  Map<String, String> persistedProperties() {
    return expectedTopology
        .flatMap(t -> t.configs)
        .orElseGet(HashMap::new);
  }

  public Map<String, Object> properties() {
    return properties;
  }

  public List<String> statements() {
    return statements;
  }

  @SuppressWarnings("unchecked")
  void processInput(final TopologyTestDriverContainer testDriver,
      final SchemaRegistryClient schemaRegistryClient) {
    inputRecords
        .stream().filter(record -> testDriver.getSourceTopics().contains(record.topic.getName()))
        .forEach(
        r -> testDriver.getTopologyTestDriver().pipeInput(
            new ConsumerRecordFactory<>(
                r.keySerializer(),
                r.topic.getSerializer(schemaRegistryClient)
            ).create(r.topic.name, r.key(), r.value, r.timestamp)
        )
    );
  }

  public static void printOutput(final TopologyTestDriverContainer topologyTestDriverContainer,
      final SchemaRegistryClient schemaRegistryClient) {
    for (int i = 0; i < 1; i++) {
      final ProducerRecord record = topologyTestDriverContainer.getTopologyTestDriver().readOutput(
          topologyTestDriverContainer.getSinkTopics().iterator().next(),
          new StringSerdeSupplier().getDeserializer(schemaRegistryClient),
          new StringDeserializer());
      System.out.println(record);
    }
  }

  @SuppressWarnings("unchecked")
  void verifyOutput(final TopologyTestDriverContainer testDriver,
      final SchemaRegistryClient schemaRegistryClient) {
    if (isAnyExceptionExpected()) {
      failDueToMissingException();
    }

    int idx = -1;
    try {
      for (idx = 0; idx < outputRecords.size(); idx++) {
        final Record expectedOutput = outputRecords.get(idx);

        final ProducerRecord record = testDriver.getTopologyTestDriver().readOutput(
            expectedOutput.topic.name,
            expectedOutput.keyDeserializer(),
            expectedOutput.topic.getDeserializer(schemaRegistryClient));

        if (record == null) {
          throw new AssertionError("No record received");
        }

        OutputVerifier.compareKeyValueTimestamp(
            record,
            expectedOutput.key(),
            expectedOutput.value,
            expectedOutput.timestamp);
      }
    } catch (final AssertionError assertionError) {
      final String rowMsg = idx == -1 ? "" : " while processing output row " + idx;
      final String topicMsg = idx == -1 ? "" : " topic: " + outputRecords.get(idx).topic.name;
      throw new AssertionError("TestCase name: "
          + name
          + " in file: " + testPath
          + " failed" + rowMsg + topicMsg + " due to: "
          + assertionError.getMessage(), assertionError);
    }
  }

  void initializeTopics(final ServiceContext serviceContext) {
    for (final Topic topic : topics) {
      serviceContext.getTopicClient().createTopic(
          topic.getName(),
          topic.numPartitions,
          (short) topic.replicas);

      topic.getSchema()
          .ifPresent(schema -> {
            try {
              serviceContext.getSchemaRegistryClient()
                  .register(topic.getName() + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX, schema);
            } catch (final Exception e) {
              throw new RuntimeException(e);
            }
          });
    }
  }

  void verifyMetastore(final MetaStore metaStore) {
    postConditions.verify(metaStore);
  }

  void verifyTopology() {
    expectedTopology.ifPresent(expected -> {
      assertThat("Generated topology differs from that built by previous versions of KSQL"
              + " - this likely means there is a non-backwards compatible change.\n"
              + "THIS IS BAD!",
          generatedTopology, is(expected.topology));

      expected.schemas.ifPresent(schemas -> {
        assertThat("Schemas used by topology differ from those used by previous versions"
                + " of KSQL - this likely means there is a non-backwards compatible change.\n"
                + "THIS IS BAD!",
            generatedSchemas, is(schemas));
      });
    });
  }

  boolean isAnyExceptionExpected() {
    return !expectedException.matchers.isEmpty();
  }

  private void failDueToMissingException() {
    final String expectation = StringDescription.toString(expectedException.build());
    final String message = "Expected test to throw" + expectation;
    fail(message);
  }

  void handleException(final RuntimeException e) {
    if (isAnyExceptionExpected()) {
      assertThat(e, isThrowable(expectedException.build()));
    } else {
      throw e;
    }
  }
}