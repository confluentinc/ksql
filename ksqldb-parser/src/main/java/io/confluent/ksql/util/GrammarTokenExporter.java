/*
 * Copyright 2019 Confluent Inc.
 *
 * Licensed under the Confluent Community License (the "License"; you may not use
 * this file except in compliance with the License. You may obtain a copy of the
 * License at
 *
 * http://www.confluent.io/confluent-community-license
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations under the License.
 */

package io.confluent.ksql.util;

import io.confluent.ksql.parser.SqlBaseLexer;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.regex.Pattern;
import java.util.stream.Collectors;

public final class GrammarTokenExporter {
  private GrammarTokenExporter() {
  }

  public static void main(final String[] args) throws IOException {
    final List<String> tokens = GrammarTokenExporter.getTokens();
    final List<String> tokensWithQuotes = new ArrayList<>();
    for (String token:tokens) {
      tokensWithQuotes.add("'" + token + "'");
    }
    System.out.println(tokensWithQuotes);
  }

  private static final Pattern pattern = Pattern.compile("(T__[0-9]|\\w+_\\w+|UNRECOGNIZED)");

  public static List<String> getTokens() {
    final List<String> tokens = new ArrayList<>(Arrays.asList(SqlBaseLexer.ruleNames));
    return tokens.stream().filter(pattern.asPredicate().negate()).collect(Collectors.toList());
  }
}
