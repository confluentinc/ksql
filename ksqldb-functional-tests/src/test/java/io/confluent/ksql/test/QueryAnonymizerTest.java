/*
 * Copyright 2021 Confluent Inc.
 * Licensed under the Confluent Community License (the "License"); you may not use
 * this file except in compliance with the License.  You may obtain a copy of the
 * License at
 * http://www.confluent.io/confluent-community-license
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations under the License.
 */

package io.confluent.ksql.test;

import static java.nio.charset.StandardCharsets.UTF_8;

import io.confluent.ksql.engine.rewrite.QueryAnonymizer;
import io.confluent.ksql.test.QueryTranslationTest.QttTestFile;
import io.confluent.ksql.test.loader.JsonTestLoader;
import io.confluent.ksql.test.tools.TestCase;
import io.confluent.ksql.util.GrammarTokenExporter;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.Arrays;
import java.util.Collection;
import java.util.Collections;
import java.util.HashSet;
import java.util.List;
import java.util.Set;
import java.util.stream.Collectors;
import org.approvaltests.Approvals;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import org.junit.experimental.runners.Enclosed;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

@RunWith(Enclosed.class)
public class QueryAnonymizerTest {
  private static final Path QUERIES_TO_ANONYMIZE_PATH =
      Paths.get("src/test/java/io/confluent/ksql/test/QueriesToAnonymizeTest.txt");
  private static final Path QUERY_VALIDATION_TEST_DIR = Paths.get("query-validation-tests");
  private final QueryAnonymizer anon = new QueryAnonymizer();

  @Test
  public void queriesAreAnonymizedCorrectly() throws Exception {
    // Given:
    StringBuilder statements = new StringBuilder();

    String line;
    try (BufferedReader reader = new BufferedReader(
        new InputStreamReader(Files.newInputStream(Paths.get(QUERIES_TO_ANONYMIZE_PATH.toString())), UTF_8))) {
      while ((line = reader.readLine()) != null) {
        statements.append(line);
      }
    }

    // When:
    final String anonStatementSelection = anon.anonymize(statements.toString());

    // Assert:
    Approvals.verify(anonStatementSelection);
  }

  private static JsonTestLoader<TestCase> testFileLoader() {
    return JsonTestLoader.of(QUERY_VALIDATION_TEST_DIR, QttTestFile.class);
  }

  @RunWith(Parameterized.class)
  public static class AnonQuerySetIntersectionTestClass {
    private List<String> sqlTokens;
    private final QueryAnonymizer anon = new QueryAnonymizer();
    private final String statement;

    public AnonQuerySetIntersectionTestClass(final String statement) {
      this.statement = statement;
    }

    @Before
    public void setUp() {
      sqlTokens = GrammarTokenExporter.getTokens();
      sqlTokens.addAll(Arrays.asList("INT", "DOUBLE", "VARCHAR", "BOOLEAN", "BIGINT", "BYTES",
          "*"));
    }

    @Parameterized.Parameters
    public static Collection<String> input() {
      return testFileLoader().load()
          .filter(statement -> !statement.expectedException().isPresent())
          .map(TestCase::statements)
          .flatMap(Collection::stream)
          .collect(Collectors.toSet());
    }

    @Test
    public void anonQuerySplitByWordsHasOnlyTokensInSetIntersectionWithQuery() {
      final String anonStatement = anon.anonymize(statement);
      final Set<String> statementWord = new HashSet<>(Arrays.asList(statement.split("[\\s\\(\\)<>,;]")));
      final Set<String> anonStatementWord = new HashSet<>(Arrays.asList(anonStatement.split("[\\s\\(\\)<>,;]")));
      final Set<String> intersection = new HashSet<>(statementWord);
      intersection.retainAll(anonStatementWord);

      // Assert:
      intersection.removeAll(sqlTokens);
      intersection.remove("");
      Assert.assertEquals(Collections.emptySet(), intersection);
    }
  }
}
