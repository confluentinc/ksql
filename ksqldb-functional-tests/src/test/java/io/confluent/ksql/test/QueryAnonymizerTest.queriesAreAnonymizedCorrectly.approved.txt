CREATE TABLE table1 (column1 INT PRIMARY KEY, column2 INT, column3 INT) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE TABLE table2 (column1 INT PRIMARY KEY, column2 BIGINT) WITH (KAFKA_TOPIC='[string]', KEY_FORMAT='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream1 (column1 INT KEY, column2 INT) WITH (KAFKA_TOPIC='[string]', KEY_FORMAT='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream2 (column1 BIGINT KEY, column9 STRING) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream3 (column1 BIGINT KEY, column9 VARCHAR, column12 BIGINT) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream4 (column1 INT KEY, column14 STRING) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream5 (column15 STRING KEY, column16 BIGINT, column17 VARCHAR) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream6 (column15 STRING KEY, column19 VARCHAR) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE TABLE table2 (column1 BIGINT PRIMARY KEY, column9 VARCHAR) WITH (KAFKA_TOPIC='[string]', KEY_FORMAT='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream7 (column1 STRING KEY, column23 DOUBLE, column24 INT) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream1 (column1 INT KEY, column15 INT, column2 BIGINT) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE TABLE table4 (column1 BIGINT PRIMARY KEY, column9 VARCHAR, column12 BIGINT) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream7 (column1 STRING KEY, column23 STRING, column24 INT) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream1 (column1 INT KEY, column2 BIGINT) WITH (KAFKA_TOPIC='[string]', KEY_FORMAT='[string]', VALUE_FORMAT='[string]');
CREATE TABLE table5 (column1 INT PRIMARY KEY, column37 STRING) WITH (KAFKA_TOPIC='[string]', KEY_FORMAT='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream11 (column15 STRING, column19 VARCHAR) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream7 (column1 STRING KEY, column23 ARRAY<INT>, column24 INT) WITH (KAFKA_TOPIC='[string]', FORMAT='[string]');
CREATE STREAM stream3 (column1 BIGINT KEY, column9 VARCHAR, column12 BIGINT) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream14 (column1 STRING KEY, column47 DECIMAL) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream3 (column1 BIGINT KEY, column9 VARCHAR, column12 BIGINT) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream4 (column15 STRING KEY, column1 VARCHAR) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE TABLE table4 (column1 BIGINT PRIMARY KEY, column9 VARCHAR, column12 BIGINT) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream1 (column1 BIGINT KEY, column9 VARCHAR, column58 BIGINT) WITH (TIMESTAMP='[string]', KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream7 (column1 STRING KEY, column23 BIGINT, column24 INT) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream3 (column1 BIGINT KEY, column9 VARCHAR, column12 BIGINT) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream1 (column1 BIGINT KEY, column9 STRING) WITH (WRAP_SINGLE_VALUE='false', KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream1 WITH (KAFKA_TOPIC='[string]', FORMAT='[string]');
CREATE TABLE table2 (column1 BIGINT PRIMARY KEY, column9 STRING) WITH (WRAP_SINGLE_VALUE='false', KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream7 (column16 INT KEY, column17 INT, column71 INT) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream7 (column1 STRING KEY, column23 STRUCT<INT, STRING>, column24 INT) WITH (KAFKA_TOPIC='[string]', FORMAT='[string]');
CREATE TABLE table8 (column1 VARCHAR PRIMARY KEY, column76 VARCHAR) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream1 (column1 INT KEY, column2 BIGINT) WITH (KAFKA_TOPIC='[string]', KEY_FORMAT='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream11 (column15 STRING KEY, column19 VARCHAR) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream7 (column1 INT KEY, column82 BIGINT) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream3 (column83 BIGINT KEY, column9 VARCHAR) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream1 (column1 INT KEY, column2 INT) WITH (KAFKA_TOPIC='[string]', KEY_FORMAT='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream14 (column15 STRING KEY, column88 ARRAY<VARCHAR>) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream30 (column15 STRING KEY, column90 BIGINT) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream31 (column91 INT KEY, column92 INT) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE TABLE table9 (column1 BIGINT PRIMARY KEY, column9 VARCHAR, column58 BIGINT) WITH (TIMESTAMP='[string]', KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream1 (column1 INT KEY, column47 BIGINT) WITH (KAFKA_TOPIC='[string]', KEY_FORMAT='[string]', VALUE_FORMAT='[string]', WINDOW_TYPE='[string]', WINDOW_SIZE='[string]');
CREATE TABLE table2 (column1 BIGINT PRIMARY KEY, column9 STRING) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream7 (column1 STRING KEY, column23 INT, column24 INT) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream3 (column15 STRING KEY, column1 BIGINT, column9 VARCHAR, column12 BIGINT) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream1 (column1 INT KEY, column2 BIGINT) WITH (KAFKA_TOPIC='[string]', KEY_FORMAT='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream7 (column1 STRING KEY, column23 BIGINT, column24 INT) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream1 (column1 INT KEY, column15 INT, column2 BIGINT) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream1 (column16 INT KEY, column17 INT) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
CREATE TABLE table2 (column1 INT PRIMARY KEY, column2 BIGINT) WITH (KAFKA_TOPIC='[string]', KEY_FORMAT='[string]', VALUE_FORMAT='[string]');
CREATE STREAM stream1 (column1 INT KEY, column16 BIGINT, column17 STRUCT<INT>) WITH (KAFKA_TOPIC='[string]', VALUE_FORMAT='[string]');
EXPLAIN SELECT * FROM source1;
DROP STREAM IF EXISTS stream6;
INSERT INTO stream41 SELECT column122, column123, column124 FROM source2 INNER JOIN source3 WITHIN '0' SECOND ON anonKey1=anonKey2;
DROP STREAM stream6;
INSERT INTO stream43 SELECT column124, column126 FROM source4;
INSERT INTO stream44 SELECT * FROM source4;
INSERT INTO stream41 SELECT * FROM source3;
INSERT INTO stream41 SELECT column122, column128, column129 FROM source2 INNER JOIN source3 WITHIN '0' SECOND ON anonKey1=anonKey2;
DROP STREAM IF EXISTS stream6;